### PR TITLE
add the publishing to gpr and release pipeline

### PR DESCRIPTION
**Pull request by vahidlazio**
* create release per tag created in the main branch.
* we can depend on openfeature sdk in the provider (on any branch/commit as following using jitpack) after merging this :

```
api("com.github.spotify:openfeature-kotlin-sdk:[BRANCH_NAME]-SNAPSHOT") // or commit hash
```
or on a release 
```
api("com.github.spotify:openfeature-kotlin-sdk")
```

### CODE REVIEW

File: OpenFeature/build.gradle.kts [hunk 1]
```diff
@@ -48,16 +45,53 @@ dependencies {
...
-            groupId = "dev.openfeature"
-            artifactId = "kotlin-sdk"
-            version = "0.0.1-SNAPSHOT"
+afterEvaluate {
+    publishing {
+        publications {
+            register<MavenPublication>("release") {
+                groupId = "dev.openfeature"
+                artifactId = "kotlin-sdk"
+                version = "0.0.1-SNAPSHOT"
```
Comment (nicklasl) [inline OpenFeature/build.gradle.kts:54]: Is this correct? shouldn't we pull this from somewhere? The tag info for example?

Comment (nicklasl) [reply OpenFeature/build.gradle.kts:54]: Yeah, sounds good. Just a note that we have the version duplicated both here and in `android { defaultConfig {} }`

Comment (fabriziodemaria) [reply OpenFeature/build.gradle.kts:52]: +1 on syncing Github release versions with Maven versions, once we get closer to an official Maven distribution

**Review by fabriziodemaria — APPROVED** [review]
I have little knowledge of `jitpack`, but my understanding is that we are going to generate `aar` libs to be attached to our draft releases here on GitHub. None of this is impacting the more "official" Maven publications.
As long as the above is correct, looks good to me 👍

**Comment by vahidlazio** [comment]
@fabriziodemaria true. btw, it's important to not mistake jitpack with the create release step. releases of the github is a different thing than jitpack. jitpack is just a tool that lets us depend on a jvm/android lib in github. basically they do all the local checkout and localMaven stuff internally.

**Pull request merged**